code/game/turfs/simulated/floor_types.dm

```diff
@@ -345,6 +348,9 @@
 							var/turf/simulated/floor/FF = get_step(src,direction)
 							FF.update_icon() //so siding get updated properly
 
+/turf/simulated/floor/carpet/attack_construct(mob/user as mob)//carpets are cool

```

Comment:
If you want them to not convert carpets you should put a check in turf/simulated/floor for the carpet tile type, any placed carpets shouldn't have this typepath